pycbc/events/single.py

```diff
@@ -31,31 +65,69 @@ def from_cli(cls, args, ifo):
         return cls(
            ifo, newsnr_threshold=args.single_newsnr_threshold[ifo],
            reduced_chisq_threshold=args.single_reduced_chisq_threshold[ifo],
-           fixed_ifar=args.single_fixed_ifar[ifo],
            duration_threshold=args.single_duration_threshold[ifo],
+           fixed_ifar=args.single_fixed_ifar,
+           fit_file=args.single_fit_file,
+           sngl_ifar_est_dist=args.sngl_ifar_est_dist[ifo]
            )
 
-    def check(self, triggers, data_reader):
+    def check(self, trigs, data_reader):
         """ Look for a single detector trigger that passes the thresholds in
         the current data.
         """
-        if len(triggers['snr']) == 0:
-            return None
-
-        i = triggers['snr'].argmax()
-        # This uses the pycbc live convention of chisq always meaning the
-        # reduced chisq.
-        rchisq = triggers['chisq'][i]
-        nsnr = ranking.newsnr(triggers['snr'][i], rchisq)
-        dur = triggers['template_duration'][i]
-
-        if nsnr > self.newsnr_threshold and \
-                rchisq < self.reduced_chisq_threshold and \
-                dur > self.duration_threshold:
-            fake_coinc = {'foreground/%s/%s' % (self.ifo, k): triggers[k][i]
-                          for k in triggers}
+        trigsremain = True
+        if len(trigs['snr']) == 0:
+            trigsremain = False
```

Comment:
Hmm, this function still smells. What happens if this first `if` succeeds? The desired behavior is that the function returns `None`, right? With the current code, `trigsremain` will be set to `False`. Then every other `if` in the function will fail, and we will get to the final `return fake_coinc`. But `fake_coinc` will not be defined at that point.

Why not just returning `None` straight away here, instead of having `trigsremain` and the various `if` statements later on? I.e. every time you do a check that fails, just `return None` at that point. Exit the function as early as you can. You also save a lot of indentation.

Reply:
Nevermind, I was wrong in my previous comment. I will actually work. But this looks unnecessarily complex, so my second comment still stands :)

Reply:
Yeah that was what I wanted to do, and what it originally did, but having quite a few return statements made codeclimate complain

Reply:
Weird. It seems to me that having multiple returns would lead to a far more readable code in this case. But who am I to disagree with the codeclimate overlords...

Reply:
So this was fixed in the end by solving the below issue - I had been using 5 returns when the maximum is 4 - it should be more readable now